src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -52,65 +54,88 @@ internal class WorkloadUpdateCommand : InstallingWorkloadCommand
-                }
-                catch (Exception e)
+                var rollbackContents = new Dictionary<string, string>();
+                foreach (var (id, version, featureBand) in rollbackFileContents)
                 {
-                    throw new GracefulException(string.Format(LocalizableStrings.WorkloadCacheDownloadFailed, e.Message), e, isUserError: false);
+                    rollbackContents[id.ToString()] = $"{version}/{featureBand}";
                 }
-            }
-            else if (_printDownloadLinkOnly)
-            {
-                var packageUrls = GetUpdatablePackageUrlsAsync(_includePreviews).GetAwaiter().GetResult();
-                Reporter.WriteLine("==allPackageLinksJsonOutputStart==");
-                Reporter.WriteLine(JsonSerializer.Serialize(packageUrls, new JsonSerializerOptions() { WriteIndented = true }));
-                Reporter.WriteLine("==allPackageLinksJsonOutputEnd==");
-            }
-            else if (_adManifestOnlyOption)
-            {
-                _workloadManifestUpdater.UpdateAdvertisingManifestsAsync(_includePreviews, string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption)).Wait();
-                Reporter.WriteLine();
-                Reporter.WriteLine(LocalizableStrings.WorkloadUpdateAdManifestsSucceeded);
-            }
-            else if (_printRollbackDefinitionOnly)
-            {
-                var workloadSet = WorkloadSet.FromManifests(_workloadResolver.GetInstalledManifests());
 
-                Reporter.WriteLine("==workloadRollbackDefinitionJsonOutputStart==");
-                Reporter.WriteLine(workloadSet.ToJson());
-                Reporter.WriteLine("==workloadRollbackDefinitionJsonOutputEnd==");
+                recorder.HistoryRecord.RollbackFileContents = rollbackContents;
             }
-            else
+
+            try
             {
-                try
+                if (!string.IsNullOrWhiteSpace(_downloadToCacheOption))
+                {
+                    try
+                    {
+                        recorder.Run(() => DownloadToOfflineCacheAsync(new DirectoryPath(_downloadToCacheOption), _includePreviews).Wait());
+                    }
+                    catch (Exception e)
+                    {
+                        throw new GracefulException(string.Format(LocalizableStrings.WorkloadCacheDownloadFailed, e.Message), e, isUserError: false);
+                    }
+                }
+                else if (_printDownloadLinkOnly)
+                {
+                    var packageUrls = GetUpdatablePackageUrlsAsync(_includePreviews).GetAwaiter().GetResult();
+                    Reporter.WriteLine("==allPackageLinksJsonOutputStart==");
+                    Reporter.WriteLine(JsonSerializer.Serialize(packageUrls, new JsonSerializerOptions() { WriteIndented = true }));
+                    Reporter.WriteLine("==allPackageLinksJsonOutputEnd==");
```

Comment:
Isn't this duplicate code from `WorkloadInstallCommand`?

Reply:
Yeah; I can probably merge those 🙂